syntax = "proto3";

import "common.proto";
import "material.proto";

package OGS;

message ElementData {
    uint64 element_id = 1;  // Element id
    uint32 n_integration_points = 2;    // Length of the following arrays.

    repeated double integration_point_coordinates = 3;
    oneof ipdata {
        SmallDeformation small_deformation = 10;
        SmallDeformationNonlocal small_deformation_nonlocal = 11;
    }
}

// Integration point data (for all integration points) common to mechanics
// process
message SmallDeformationCommon {
    repeated KelvinVector sigma = 1;
    repeated KelvinVector eps = 2;
    repeated MaterialState material_state = 3;
}

message SmallDeformation {
    SmallDeformationCommon common = 1;
}

message SmallDeformationNonlocal {
    SmallDeformationCommon common = 1;
    repeated double nonlocal_damage = 2;
    repeated double nonlocal_length = 3;
}
